warehouse/deltalake/deltalake.go

```diff
@@ -868,11 +868,9 @@ func (dl *HandleT) DropTable(tableName string) (err error) {
 	return
 }
 
-// AddColumn adds column for column name and type
-func (dl *HandleT) AddColumn(name, columnName, columnType string) (err error) {
-	tableName := fmt.Sprintf(`%s.%s`, dl.Namespace, name)
-	sqlStatement := fmt.Sprintf(`ALTER TABLE %v ADD COLUMNS ( %s %s );`, tableName, columnName, getDeltaLakeDataType(columnType))
-	pkgLogger.Infof("%s Adding column in delta lake with SQL:%v", dl.GetLogIdentifier(tableName, columnName), sqlStatement)
+func (dl *HandleT) AddColumns(tableName string, columnsInfo warehouseutils.ColumnsInto) (err error) {
+	sqlStatement := addColumnsSQLStatement(dl.Namespace, tableName, columnsInfo)
```

Comment:
I am not sure if the `addColumnsSQLStatement` is necessary since it need only from this method, it would be better for readability for this code to be in one place:

```suggestion
	format := func(_ int, columnInfo warehouseutils.ColumnInfoT) string {
		return fmt.Sprintf(`%s %s`, columnInfo.Name, getDeltaLakeDataType(columnInfo.Type))
	}
	sqlStatement =  fmt.Sprintf(`
		ALTER TABLE
		%s.%s
		ADD COLUMNS ( %s );`,
		namespace,
		tableName,
		columnsInfo.JoinColumns(format, ","),
	)
```

The same comments applies in other files as well.

Reply:
In favor of this. Once @abhimanyubabbar is back, we will finalize on this.